libs/auth-api-lib/src/lib/delegations/delegations-incoming.service.ts

```diff
@@ -187,10 +189,14 @@ export class DelegationsIncomingService {
 
     const delegationSets = await Promise.all(delegationPromises)
 
-    const delegations = ([] as MergedDelegationDTO[])
+    let delegations = ([] as MergedDelegationDTO[])
       .concat(...delegationSets)
       .filter((delegation) => delegation.fromNationalId !== user.nationalId)
 
+    if (otherUser) {
+      delegations = delegations.filter((d) => d.fromNationalId === otherUser)
+    }
+
     const mergedDelegationMap = delegations.reduce(
```

Comment:
I would really like to merge scopes (and types) across delegations in a smarter way to support interesting edge/use cases, but it's not a priority for now.